crates/bdk/src/wallet/mod.rs

```diff
@@ -1980,6 +1992,15 @@ impl<D> Wallet<D> {
             }
         }
 
+        if finished && sign_options.remove_taproot_extras {
+            for output in &mut psbt.outputs {
+                // Only clear derivations for taproot outputs.
+                if output.tap_internal_key.is_some() {
+                    output.bip32_derivation.clear();
+                }
```

Comment:
```suggestion
                output.bip32_derivation.clear();
```

I think we should clear this for all finalized outputs.